kuwala/core/canvas/src/pages/DataSourceConfiguration.js

```diff
@@ -0,0 +1,112 @@
+import React from "react";
+import Header from "../components/Header";
+import {useLocation, Link} from "react-router-dom";
+import {useStoreState} from "easy-peasy";
+
+export default () => {
+
+    const location = useLocation()
+    const dataIndex = location.state.index
+    const { dataSource } = useStoreState((state) => state.canvas);
+    const selectedSource = dataSource[dataIndex]
+
+    const renderDataSourceConfig = () => {
+        if(!selectedSource) {
+            return (
+                <div>
+                    Undefined data source, something is wrong.
+                </div>
+            )
+        }else {
+            console.log(selectedSource)
+            return (
+                <div className={'flex flex-col bg-white px-8 py-4 rounded-lg h-full'}>
+                    {selectedSource.connection_parameters.map((e,i) => {
+                        return (
+                            <div className={'flex flex-row h-full w-full items-center'}>
+                                <div className={'w-1/6'}>
+                                    <span className={'text-lg capitalize font-bold'}>
+                                        {e.id}
+                                    </span>
+                                </div>
+                                <div className={'w-5/6'}>
+                                    <input
+                                        type="search"
+                                        className="w-full px-4 py-2 border-2 border-kuwala-green text-gray-800 capitalize rounded-lg focus:outline-none"
+                                        placeholder={e.id}
+                                    />
+                                </div>
+                            </div>
+                        )
+                    })}
+                    <div className={'flex flex-row justify-between mt-6'}>
+                        <button
+                            className={'bg-white border-2 border-kuwala-green rounded-md px-3 py-2 mt-4 hover:text-stone-300'}
+                        >
+                            <span className={'text-kuwala-green'}>Check Connection</span>
+                        </button>
+                        <button
+                            className={'bg-kuwala-green rounded-md px-3 py-2 mt-4 hover:text-stone-300'}
+                        >
+                            <span className={'text-white'}>Save</span>
+                        </button>
```

Comment:
The save button should be disabled as long as not all configurations have been made

Reply:
This one is still WIP and possibly will change the form into Formik https://www.npmjs.com/package/formik